packages/amplify_core/lib/src/types/auth/attribute/auth_user_attribute_key.dart

```diff
@@ -15,7 +16,11 @@ abstract class AuthUserAttributeKey
     with AWSSerializable<String>
     implements Comparable<AuthUserAttributeKey> {
   /// {@macro amplify_core.auth_user_attribute_key}
-  const AuthUserAttributeKey();
+  const factory AuthUserAttributeKey(String key) = _AuthUserAttributeKey;
```

Comment:
I'm not sure I understand the update here. Why add the factory constructor? I see it used in the converter, but you could use _AuthUserAttributeKey directly there, no?

Do we want to expose this for customers?

Reply:
Good catch - removed